### ISSUE TITLE
hw13

### ISSUE THREAD

**Issue by zpcohen** [kingcohn1/stat679work#2]
@cecileane

**Comment by cecileane** [comment]
Thank you Zach! This is my way of asking Xuchun (@xuchun725) and Ruifeng (@ruifengchen) to try your script and provide feedback on it. 

Looking quickly at your code, it's not clear what files they should look at. I suggest that you give them directions about where to find the correct code, in some readme file.

**Comment by zpcohen** [comment]
Ok, I created a .md file [here](https://github.com/kingcohn1/stat679work/blob/master/ReadmeHW13.md).

**Comment by zpcohen** [comment]
Dear All, 
I have cleaned up my github repository for your convenience and my sanity. You can now 
`git pull origin master` to get my most recent commit. Thanks and I look forward to getting your feedback!
-Z

**Comment by ruifengchen** [comment]
Hi Zach,
Excellent job! some feedback below.

I think that these “set” functions at the top are very good which I should try in my own work. You have used a lot of useful commands like “grep”, “cut”, “basename” and “sed”, these are the parts of significance we have learnt and what the course hopes us to do, thus carries big weight! Also, these comments are really detailed. Furthermore, you mostly use “sed” first and then “grep”, which I believe is very nice.

Data is almost good in the result csv file, but unfortunately, columns h and CPUtime are missing. I believe it is a small part of misunderstanding but please be careful with the question next time, we in fact should do exercise3 continuing with exercise1&2, thus the last two columns called h and CPUtime should be concluded in the result csv file.

variable filename in your script is called “i”, if possible, please try “filename” or something else which is more readable.

You made a nice format of your readme file, thanks for changing on time. A nice organization is very helpful for me to read and for you to get back in the future.

Fantastic job again, but if possible, please change everything quickly next time, it would be more convenient for me, thanks! Anyway, good job.

Thank you!
Ruifeng

**Comment by xuchun725** [comment]
Hi Zach,

Congratulations on solving the syncing issue! Writing down what problem you were faced with and how you fixed it in `readme.md` file is a really good idea and I will do this in my further project. I also like other parts in your `readme.md` file, such as how you got the data, how you created a local repository and even your laptop system and library versions. It would be better if you say which file your script created.

Your code for ex3 runs well except missing two columns we generated in ex2. You have a smart use of `basename`, `for` loop, `if` statement, _arithmetic expansion_ and _parameter expansion_ and get much practice with `grep`, `sed`, `cut` and _regular expressions_. Inline comments make it really easy to follow. `set` commands at the beginning of the script make your script safer, which is what I should learn from you.

To make your script look better, I think the indentation should agree with the logic nesting depth. Please pay attention to the indentation of nested loops.
There are two commands I don't know what they are for: `echo $analysis,$Nruns,$Nfail,$fabs,$frel,$xabs,$xrel,$seed`(line 20) and `echo $l,$a`(line 32). Are they just redundant or have some other effects?
The column names (eg,`nRuns`) and the variable names (eg, `Nruns`) are not consistent. Although it's not a big issue, I think keeping them consistent can make a beautiful coding style.

Your commit messages are informative. It seems neither `v1.2` nor `v1.22` is your final version of ex1&2. If you want to delete a tag, you can use `git tag -d <tag name>` and then tag the commit which was modified based on professor's ex1&2 feedback.

Thanks for sharing your scripts! It's a great work and I learned a lot from your scripts.

Best,
Xuchun

**Comment by zpcohen** [comment]
Great, thank you for constructive feedback! Ruifeng, I took your advice and added columns for `$h` and `$CPU_time`. I also changed `$i` to `$filename`.  And Xuchen, thank you for addressing that inconsistency with my variable and column names, I've also made those changes. I will also delete the commits as you suggested. I really appreciate your experience with this and was wondering if you'd like to form a group for the upcoming project? Pablo @pgonzalez6 , another classmate of ours, and I are looking to start a group so let me know if you're available!
